benchmarks/benchmarks/sparse_linalg_lobpcg.py

```diff
@@ -66,45 +29,119 @@ def __init__(self):
-        self.A, self.B = _mikota_pair(n)
-
-        if solver == 'eigh' and n >= 512:
+        mik = MikotaPair(n)
+        mik_k = mik.k
+        mik_m = mik.m
+        self.Ac = mik_k
+        self.Aa = mik_k.toarray()
+        self.Bc = mik_m
+        self.Ba = mik_m.toarray()
+        self.Ab = mik_k.tobanded()
+        self.eigenvalues = mik.eigenvalues
+
+        if solver == 'lapack' and n > 512:
             # skip: slow, and not useful to benchmark
             raise NotImplementedError()
 
     def setup_sakurai(self, n, solver):
         self.shape = (n, n)
-        self.A, self.B, all_eigenvalues = _sakurai(n)
-        self.A_dense = self.A.A
-        self.B_dense = self.B.A
+        sakurai_obj = Sakurai(n, dtype='int')
+        self.A = sakurai_obj
+        self.Aa = sakurai_obj.toarray()
+        self.eigenvalues = sakurai_obj.eigenvalues
 
+    def setup_sakuraii(self, n, solver):
+        self.shape = (n, n)
+        sakurai_obj = Sakurai(n)
+        self.A = sakurai_obj.tobanded()
+        self.eigenvalues = sakurai_obj.eigenvalues
+    
     def time_mikota(self, n, solver):
+        def a(x):
+            return cho_solve_banded((c, False), x)
         m = 10
+        ee = self.eigenvalues[:m]
+        tol = m * n * n * n* np.finfo(float).eps
+        rng = np.random.default_rng(0)
+        X = rng.normal(size=(n, m))
         if solver == 'lobpcg':
-            X = np.random.rand(n, m)
-            X = orth(X)
-            LorU, lower = cho_factor(self.A, lower=0, overwrite_a=0)
-            M = LinearOperator(self.shape,
-                               matvec=partial(_precond, LorU, lower),
-                               matmat=partial(_precond, LorU, lower))
-            eigs, vecs = lobpcg(self.A, X, self.B, M, tol=1e-4, maxiter=40)
+            c = cholesky_banded(self.Ab.astype(np.float32))
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+                el, _ = lobpcg(self.Ac, X, self.Bc, M=a, tol=1e-4,
+                               maxiter=40, largest=False)
+            accuracy = max(abs(ee - el) / ee)
+            assert accuracy < tol, msg
+        elif solver == 'eigsh':
+            B = LinearOperator((n, n), matvec=self.Bc, matmat=self.Bc, dtype='float64')
+            A = LinearOperator((n, n), matvec=self.Ac, matmat=self.Ac, dtype='float64')
+            c = cholesky_banded(self.Ab)
+            a_l = LinearOperator((n, n), matvec=a, matmat=a, dtype='float64')
+            ea, _ = eigsh(B, k=m, M=A, Minv=a_l, which='LA', tol=1e-4, maxiter=50,
+                          v0 = rng.normal(size=(n, 1)))
+            accuracy = max(abs(ee - np.sort(1./ea)) / ee)
+            assert accuracy < tol, msg
         else:
-            eigh(self.A, self.B, eigvals_only=True, eigvals=(0, m - 1))
+            ed, _ = eigh(self.Aa, self.Ba, subset_by_index=(0, m - 1))
+            accuracy = max(abs(ee - ed) / ee)
+            assert accuracy < 0. * tol, msg
```

Comment:
What's the intention here of something that always fails?

Reply:
Just temp, testing the assert msg, which seems to work, so go back to normal now